core/transaction_pool.go

```diff
@@ -232,8 +230,15 @@ func (pool *TxPool) validateTx(tx *types.Transaction) error {
 		return ErrInsufficientFunds
 	}
 
+	// TODO since this can't be checked using block numbers (since this process
+	// is time sensetive rather than block sensetive) we must, for now, use
+	// frontier rule sets. After frontier we must change this to homestead
+	// ruleset.
+	// Right now only a small subset of transaction would be allowed that
+	// could potentially have insufficient intrinsic gas at homestead.
 	// Should supply enough intrinsic gas
-	if tx.Gas().Cmp(IntrinsicGas(tx.Data())) < 0 {
+	intrGas := IntrinsicGas(tx.Data(), MessageCreatesContract(tx), false)
+	if tx.Gas().Cmp(intrGas) < 0 {
```

Comment:
Same here really, a malicious user could DOS the network with non executable transactions after homestead if we accept them into the pool.